devicetypes/smartthings/inovelli-dimmer.src/inovelli-dimmer.groovy

```diff
@@ -0,0 +1,599 @@
+		} else {
+			state.currentPreferencesState."$it.key".status = "synced"
+		}
+	}
+	readConfigurationFromTheDevice();
+// Preferences template end
+	createChildDevice("smartthings", "Child Color Selection", "${device.deviceNetworkId}:" + 3, "LEDColorConfiguration", "LEDColorConfiguration")
+}
+
+def updated() {
+	// Device-Watch simply pings if no device events received for 32min(checkInterval)
+	sendEvent(name: "checkInterval", value: 2 * 15 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
+	// Preferences template begin
+	parameterMap.each {
+		if (isPreferenceChanged(it)) {
+			log.debug "Preference ${it.key} has been updated from value: ${state.currentPreferencesState."$it.key".value} to ${settings."$it.key"}"
+			state.currentPreferencesState."$it.key".status = "syncPending"
+			if (it.type == "boolRange") {
+				def preferenceName = it.key + "Boolean"
+
+				if (isNotNull(settings."$preferenceName")) {
+					if (!settings."$preferenceName") {
+						state.currentPreferencesState."$it.key".status = "disablePending"
+					} else if (state.currentPreferencesState."$it.key".status == "disabled") {
+						state.currentPreferencesState."$it.key".status = "syncPending"
+					}
+				} else {
+					state.currentPreferencesState."$it.key".status = "syncPending"
+				}
+			}
+		} else if (state.currentPreferencesState."$it.key".value == null) {
+			log.warn "Preference ${it.key} no. ${it.parameterNumber} has no value. Please check preference declaration for errors."
+		}
+	}
+	syncConfiguration()
+	// Preferences template end
+
+	response(refresh())
+}
+
+private readConfigurationFromTheDevice() {
+	def commands = []
+	parameterMap.each {
+		state.currentPreferencesState."$it.key".status = "reverseSyncPending"
+		commands += encap(zwave.configurationV2.configurationGet(parameterNumber: it.parameterNumber))
+	}
+	sendHubCommand(commands)
+}
+
+private syncConfiguration() {
+	def commands = []
+	log.debug"syncConfiguration ${settings}"
+	parameterMap.each {
+		if (state.currentPreferencesState."$it.key".status == "syncPending") {
+			commands += encap(zwave.configurationV2.configurationSet(scaledConfigurationValue: getCommandValue(it), parameterNumber: it.parameterNumber, size: it.size))
+			commands += encap(zwave.configurationV2.configurationGet(parameterNumber: it.parameterNumber))
+		} else if (state.currentPreferencesState."$it.key".status == "disablePending") {
+			commands += encap(zwave.configurationV2.configurationSet(scaledConfigurationValue: it.disableValue, parameterNumber: it.parameterNumber, size: it.size))
+			commands += encap(zwave.configurationV2.configurationGet(parameterNumber: it.parameterNumber))
+		}
+	}
+	sendHubCommand(commands)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.configurationv2.ConfigurationReport cmd) {
+	// Preferences template begin
+	log.debug "Configuration report: ${cmd}"
+	def preference = parameterMap.find( {it.parameterNumber == cmd.parameterNumber} )
```

Comment:
I guess I've missed some things when I was merging new code to our template, but could you fix all those small spacing issues in this method? (like `if (condition) {...}` instead of `if(condition){...}`)

Reply:
Done